packages/react/src/analytics/integrations/Guardian/__tests__/Guardian.test.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  EnvironmentTypes,
+  integrations,
+  utils,
+} from '@farfetch/blackout-analytics';
+import { type GuardianIntegrationOptions, GuardianProviders } from '../types';
+import Guardian from '../Guardian';
+
+utils.logger.error = jest.fn();
+utils.logger.warn = jest.fn();
+
+const spyError = jest.spyOn(utils.logger, 'error');
+const spyWarn = jest.spyOn(utils.logger, 'warn');
+
+describe('Guardian', () => {
+  const Forter = require('../providers').Forter;
+  const Riskified = require('../providers').Riskified;
+
+  describe('Guardian integration', () => {
+    it('Should extend the core integration class', () => {
+      expect(Guardian.prototype).toBeInstanceOf(integrations.Integration);
+    });
+
+    it('Should be an integration that loads independently of user consent', () => {
+      expect(Guardian.shouldLoad()).toBe(true);
+      // @ts-expect-error
+      expect(Guardian.shouldLoad({ marketing: true })).toBe(true);
+    });
+  });
+
+  describe('Guardian instance', () => {
+    const spyForter = jest.spyOn(Forter, 'createInstance');
+    const spyRiskified = jest.spyOn(Riskified, 'createInstance');
+
+    beforeEach(jest.clearAllMocks);
+
+    it('should not load all instances by default', () => {
+      // @ts-expect-error
+      const guardianInstance = Guardian.createInstance() as Guardian;
+
+      expect(guardianInstance.activeIntegrations).toEqual([
+        GuardianProviders.Forter,
+        GuardianProviders.Riskified,
+      ]);
+
+      expect(guardianInstance.riskifiedInstance).toBeInstanceOf(
+        integrations.Integration,
```

Comment:
Shouldn't this test if it is an instance of Riskified? Same for the test of `forterInstance` below.

Reply:
I was simply verifying if all instances are automatically loaded. I believe it would be enought to ensure that instances designated as `riskified` or `forter` are indeed instances of `Integration` which is the class where each one extend.

So I don't see here too much importance to validate the exact class type in Guardian, but I will take care to include an additional test for this purpose.

Reply:
Just for information: The test should indeed check for the correct instance type, even though typescript could catch these errors but it might not if for example, the types of both `Forter` and `Riskified` classes are compatible (for example if `Riskified` was a subclass of `Forter` or they had the same property names and types). If that were the case, the test would not detect a bug if the code set the Forter instance to a Riskified instance and vice-versa.

Reply:
As I tried to explain, in this specific test I only are validating if instances are set or not depending of activeIntegration property.
Then I added an additional case, to really test if forter instance are really a forter, and same for riskified.

So in my opinion, this could be done in two different tests for different proposes (one for validating if instance aren't undefined, and another to validate the type of each instance), instead of mixing in a single test.

Reply:
Imagine that you have a bug in your code that if `riskified` option is passed, you are creating a `forter` instance instead. The test the way it is now, would not catch it. I do not see the need of having a separate test for the type of instances when you can have this test do it and would catch more errors. The test you added for that is only testing when no options are being sent (which is not valid now since both providers will need to receive options in order to be instantiated).

Reply:
Updated